src/finch/tensor.py

```diff
@@ -1,39 +1,45 @@
-from abc import abstractmethod
-from typing import Union
+from typing import Any, Optional
 
 import numpy as np
 import juliacall as jc
 
 from .julia import jl
-
-
-class _Display:
-    def __repr__(self):
-        return jl.sprint(jl.show, self._obj)
-
-    def __str__(self):
-        return jl.sprint(jl.show, jl.MIME("text/plain"), self._obj)
+from .levels import _Display, Dense, Element, Storage
+from .typing import OrderType, JuliaObj, TupleOf3Arrays
 
 
 class Tensor(_Display):
     """
     A wrapper class for Finch.Tensor and Finch.SwizzleArray.
 
-    Two constructors are supported: `lvl`+`arr`, or `jl_data`.
+    Constructors
+    ------------
+    Tensor(scipy.sparse.spmatrix)
+        Construct a Tensor out of a `scipy.sparse` object. Supported formats are: `COO`, `CSC`, and `CSR`.
+    Tensor(numpy.ndarray)
+        Construct a Tensor out of a NumPy array object. This is a no-copy operation.
+    Tensor(storage=storage)
+        Initialize a Tensor with a `storage` description. `storage` can already hold data.
+    Tensor(arr=Tensor, storage=storage)
+        Tensor created by moving `arr` contents into `storage` description.
+    Tensor(jl_data=julia_object)
+        Tensor created from a compatible raw Julia object. Must be a `SwizzleArray`.
+        This is a no-copy operation.
 
     Parameters
     ----------
-    lvl : AbstractLevel, optional
-        Levels description.
-    arr : ndarray, optional
-        NumPy array that should fill `lvl`.
+    arr : np.ndarray or scipy.sparse or Tensor, optional
+        Input to construct a Tensor. It's a no-copy operation of for NumPy and SciPy input. For Tensor
+        `storage` must also be provided.
     jl_data : Finch.SwizzleArray, optional
         Raw Julia object.
-    order : str or tuple[int, ...], optional
-        Order of the tensor. The order numbers the dimensions from the fastest to slowest.
-        The leaf nodes have mode `0` and the root node has mode `n-1`. If the tensor was square
-        of size `N`, then `N .^ order == strides`. Available options are "C" (row-major),
-        "F" (column-major), or a custom order. Default: row-major.
+    storage : Storage, optional
```

Comment:
Small request: Could we make the constructor with Storage take the storage as a positional argument? Could we do the same with jl_data?

Reply:
Done!